File: employee-service/src/main/java/com/example/employeeservice/entity/AbstractEntity.java
package com.example.employeeservice.entity;

import java.util.Objects;
import javax.persistence.Id;
import javax.persistence.MappedSuperclass;
import lombok.Getter;
import lombok.NoArgsConstructor;
import lombok.Setter;
import lombok.experimental.SuperBuilder;

@MappedSuperclass
@Getter
@Setter
@SuperBuilder
@NoArgsConstructor
public abstract class AbstractEntity<ID> {

  @Id
  protected ID id;


  @Override
  public boolean equals(Object other) {
    if (this == other) {
      return true;
    }

    if (other instanceof AbstractEntity<?> that) {
      return Objects.equals(id, that.id);
    }

    return false;
  }

  @Override
  public int hashCode() {
    return Objects.hash(id);
  }

}

File: department-service/src/main/kotlin/com/example/departmentservice/entity/AbstractEntity.kt
package com.example.departmentservice.entity

import java.util.Objects
import javax.persistence.Id
import javax.persistence.MappedSuperclass

@MappedSuperclass
abstract class AbstractEntity<ID>(
    @Id val id: ID
) {
    override fun equals(other: Any?): Boolean {
        if (this === other) return true
        if (javaClass != other?.javaClass) return false

        other as AbstractEntity<*>
        if (id != other.id) return false

        return true
    }

    override fun hashCode(): Int {
        return Objects.hashCode(id)
    }
}